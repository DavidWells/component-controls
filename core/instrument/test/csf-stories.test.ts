import { loadStoriesTests } from './loadTestFiles';

describe('csf-stories', () => {
  loadStoriesTests(undefined, ['csf', 'stories']);
});
