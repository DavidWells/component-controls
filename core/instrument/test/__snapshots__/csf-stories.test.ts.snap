// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csf-stories named-object-export.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T04:06:01.880Z,
    "dateModified": 2020-04-14T04:56:35.949Z,
    "package": "023d0d8cf3e4a1a38f08775068d9729d",
    "title": "Story",
  },
  "packages": Object {
    "023d0d8cf3e4a1a38f08775068d9729d": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.6.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "023d0d8cf3e4a1a38f08775068d9729d",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/stories/named-object-export.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.6.0",
    },
  },
  "stories": Object {},
  "transformed": "export default { title: 'Story' };
export const myStory = {
  i: 1,
};
",
}
`;

exports[`csf-stories no-arguments.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T04:06:17.604Z,
    "dateModified": 2020-04-14T04:56:08.906Z,
    "package": "2bf18b76a1b9e90d723ba13b6a10fc58",
    "title": "Story",
  },
  "packages": Object {
    "2bf18b76a1b9e90d723ba13b6a10fc58": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.6.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "2bf18b76a1b9e90d723ba13b6a10fc58",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/stories/no-arguments.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.6.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = () => {};
",
}
`;

exports[`csf-stories no-story.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T04:04:54.289Z,
    "dateModified": 2020-04-14T04:56:12.341Z,
    "package": "77d49f77a8b5ff73e715da87e9ab3277",
    "title": "Story",
  },
  "packages": Object {
    "77d49f77a8b5ff73e715da87e9ab3277": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.6.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "77d49f77a8b5ff73e715da87e9ab3277",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/stories/no-story.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.6.0",
    },
  },
  "stories": Object {},
  "transformed": "export default { title: 'Story' };
const i = 1;
",
}
`;

exports[`csf-stories props-argument.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T04:06:38.295Z,
    "dateModified": 2020-04-14T04:56:16.733Z,
    "package": "d8b5c54145ef8f8db3e19289e1c19f95",
    "title": "Story",
  },
  "packages": Object {
    "d8b5c54145ef8f8db3e19289e1c19f95": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.6.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "d8b5c54145ef8f8db3e19289e1c19f95",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/stories/props-argument.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.6.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 5,
              "line": 0,
            },
            "start": Object {
              "column": 0,
              "line": 0,
            },
          },
          "name": "props",
          "value": "props",
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 34,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "props => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = props => {};
",
}
`;

exports[`csf-stories three-levels-alias.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T04:08:27.099Z,
    "dateModified": 2020-04-14T04:56:20.303Z,
    "package": "83b0676f45ac751b689864159d82ed31",
    "title": "Story",
  },
  "packages": Object {
    "83b0676f45ac751b689864159d82ed31": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.6.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "83b0676f45ac751b689864159d82ed31",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/stories/three-levels-alias.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.6.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 31,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": undefined,
          "value": Array [
            Object {
              "loc": Object {
                "end": Object {
                  "column": 24,
                  "line": 0,
                },
                "start": Object {
                  "column": 9,
                  "line": 0,
                },
              },
              "name": "name",
              "value": Array [
                Object {
                  "loc": Object {
                    "end": Object {
                      "column": 16,
                      "line": 0,
                    },
                    "start": Object {
                      "column": 11,
                      "line": 0,
                    },
                  },
                  "name": "first",
                  "value": "first",
                },
                Object {
                  "loc": Object {
                    "end": Object {
                      "column": 22,
                      "line": 0,
                    },
                    "start": Object {
                      "column": 18,
                      "line": 0,
                    },
                  },
                  "name": "last",
                  "value": "last",
                },
              ],
            },
            Object {
              "loc": Object {
                "end": Object {
                  "column": 29,
                  "line": 0,
                },
                "start": Object {
                  "column": 26,
                  "line": 0,
                },
              },
              "name": "age",
              "value": "age",
            },
          ],
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 61,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "({ name: { first, last }, age }) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = ({ name: { first, last }, age }) => {};
",
}
`;

exports[`csf-stories two-arguments.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T04:06:59.049Z,
    "dateModified": 2020-04-14T04:56:23.651Z,
    "package": "28503fc854679acdc04bb4eccfeacbb3",
    "title": "Story",
  },
  "packages": Object {
    "28503fc854679acdc04bb4eccfeacbb3": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.6.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "28503fc854679acdc04bb4eccfeacbb3",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/stories/two-arguments.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.6.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 6,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": "props",
          "value": "props",
        },
        Object {
          "loc": Object {
            "end": Object {
              "column": 15,
              "line": 0,
            },
            "start": Object {
              "column": 8,
              "line": 0,
            },
          },
          "name": "context",
          "value": "context",
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 45,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "(props, context) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = (props, context) => {};
",
}
`;

exports[`csf-stories two-levels-alias.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T04:08:11.695Z,
    "dateModified": 2020-04-14T04:56:27.188Z,
    "package": "937b0fd35d556a5e5cfcb7fb25a785c8",
    "title": "Story",
  },
  "packages": Object {
    "937b0fd35d556a5e5cfcb7fb25a785c8": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.6.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "937b0fd35d556a5e5cfcb7fb25a785c8",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/stories/two-levels-alias.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.6.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 21,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": undefined,
          "value": Array [
            Object {
              "loc": Object {
                "end": Object {
                  "column": 14,
                  "line": 0,
                },
                "start": Object {
                  "column": 9,
                  "line": 0,
                },
              },
              "name": "name",
              "value": "MyNam",
            },
            Object {
              "loc": Object {
                "end": Object {
                  "column": 19,
                  "line": 0,
                },
                "start": Object {
                  "column": 16,
                  "line": 0,
                },
              },
              "name": "age",
              "value": "age",
            },
          ],
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 51,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "({ name: MyNam, age }) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = ({ name: MyNam, age }) => {};
",
}
`;

exports[`csf-stories two-levels-sub-arguments.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T04:07:36.418Z,
    "dateModified": 2020-04-14T04:56:30.571Z,
    "package": "6fe1b3bd8baa1d5965b447e926972dab",
    "title": "Story",
  },
  "packages": Object {
    "6fe1b3bd8baa1d5965b447e926972dab": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.6.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "6fe1b3bd8baa1d5965b447e926972dab",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/stories/two-levels-sub-arguments.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.6.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 14,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": undefined,
          "value": Array [
            Object {
              "loc": Object {
                "end": Object {
                  "column": 7,
                  "line": 0,
                },
                "start": Object {
                  "column": 3,
                  "line": 0,
                },
              },
              "name": "name",
              "value": "name",
            },
            Object {
              "loc": Object {
                "end": Object {
                  "column": 12,
                  "line": 0,
                },
                "start": Object {
                  "column": 9,
                  "line": 0,
                },
              },
              "name": "age",
              "value": "age",
            },
          ],
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 44,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "({ name, age }) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = ({ name, age }) => {};
",
}
`;

exports[`csf-stories typescript.ts 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T04:07:18.256Z,
    "dateModified": 2020-04-14T04:51:39.559Z,
    "package": "b683a7b765d6b542242ea58703bc8138",
    "title": "Story",
  },
  "packages": Object {
    "b683a7b765d6b542242ea58703bc8138": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.6.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "b683a7b765d6b542242ea58703bc8138",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/stories/typescript.ts",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.6.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 18,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": "props",
          "value": "props",
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 48,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "(props: Properties) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = (props: Properties) => {};
",
}
`;
